pkg/synchromanager/clustersynchro/resource_synchro.go

```diff
@@ -317,6 +322,12 @@ func (synchro *ResourceSynchro) OnDelete(obj interface{}) {
 		return
 	}
 
+	if !synchro.transDeleteObject {
+		synchro.pruneObject(obj.(*unstructured.Unstructured))
+		_ = synchro.queue.Delete(obj)
+		return
+	}
```

Comment:
For the sake of storage generality, it may seem necessary to call Delete with the full object, but there are some special cases where we still need to consider the `DeletedFinalStateUnknown`.

```go

func (synchro *ResourceSynchro) OnDelete(obj interface{}) {
        if !synchro.isRunnableForStorage.Load() {
                return
        }

        if d, ok := obj.(cache.DeletedFinalStateUnknown); ok {
                if obj, ok = d.Obj.(*unstructured.Unstructured); !ok {
                        namespace, name, err := cache.SplitMetaNamespaceKey(d.Key)
                        if err != nil {
                                return
                        }
                        obj = &metav1.PartialObjectMetadata{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}}
                }
        }
        if o, ok := obj.(*unstructured.Unstructured); ok {
                synchro.pruneObject(o)
        }
        _ = synchro.queue.Delete(obj)
```

Sometimes, inevitably, we can't get the full object when deleting it.